ducc/src/ducc.rs

```diff
@@ -379,7 +379,7 @@ impl Ducc {
                 ffi::DUK_TYPE_STRING => {
                     Value::String(String(self.pop_ref()))
                 },
-                ffi::DUK_TYPE_OBJECT => {
+                ffi::DUK_TYPE_OBJECT if ffi::duk_is_buffer_data(self.ctx, -1) == 0 => {
```

Comment:
This is a bit annoying, but needed since Duktape considers the standard buffers to be `ffi::DUK_TYPE_OBJECT`

Reply:
If I'm understanding the Duktape API correctly, I think you can reformat this as follows:

```rust
// ...
ffi::DUK_TYPE_OBJECT | ffi::DUK_TYPE_BUFFER => {
    if ffi::duk_is_buffer_data(self.ctx, -1) != 0 {
        // `duk_is_buffer_data` returns true if the value "is a plain buffer or any buffer object type"
        Value::Bytes(Bytes(self.pop_ref()))
    } else if ffi::duk_is_function(self.ctx, -1) != 0 {
        // ...
    } // ...
},
_ =>Value::Undefined,
// ...
```

Just a tiny optimization. This avoids calling `duk_is_buffer_data` twice when the value is indeed a standard buffer.

Reply:
That's perfect. I don't know how I missed that `ffi::DUK_TYPE_OBJECT` match arm already has the perfect placement for it 🤦